docs/maintenance-migrating.md

```diff
@@ -1,14 +1,71 @@
-2. Stop all services on the old server and make sure they won't be starting again. Execute this on the old server: `systemctl disable --now matrix*` (you might have to cd to /etc/systemd/system/ first)
-3. Copy directory `/matrix` from the old server to the new server. Make sure to preserve ownership and permissions (use `cp -p` or `rsync -ar`)!
-4. Make sure your DNS records are adjusted to point to the new server's IP address
-5. Remove old server from the `inventory/hosts` file and add new server.
-6. Run `ansible-playbook -i inventory/hosts setup.yml --tags=setup-system-user`. This will create the `matrix` user and group on the new server
-7. Because the `matrix` user and group are created dynamically on each server, the user/group ID may differ between the old and new server. We suggest that you adjust ownership of `/matrix` files manually by running this on the new server: `chown -R matrix:matrix /matrix`.
-8. Run `ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,start` to finish the installation and start all services
+This documentation explains how to migrate your Matrix services (server, client, bridges, etc.) and data from an old server to a new server.
+
+**Notes**:
+- This migration guide is applicable if you migrate from one server to another server having the same CPU architecture (e.g. both servers being `amd64`).
+
+  If you're trying to migrate between different architectures (e.g. `amd64` --> `arm64`), simply copying the complete `/matrix` directory is **not** possible as it would move the raw PostgreSQL data (`/matrix/postgres/data`) between different architectures. In this specific case, you can use the guide below as a reference, but you would also need to avoid syncing `/matrix/postgres/data` to the new host, and also dump the database on your current server and import it properly on the new server. See our [Backing up PostgreSQL](maintenance-postgres.md#backing-up-postgresql) docs for help with PostgreSQL backup/restore.
+- If you have any questions about migration or encountered an issue during migration, do not hesitate to ask for help on [our Matrix room](https://matrix.to/#/%23matrix-docker-ansible-deploy:devture.com).
+
+## Lower DNS TTL
+
+Prepare by lowering DNS TTL for your domains (`matrix.example.com`, etc.), so that DNS record changes would happen faster, leading to less downtime.
+
+## Stop services on the old server
+
+Before migrating, you need to stop all services on the old server and make sure they won't be starting again.
+
+To stop the services, go to the `matrix-docker-ansible-deploy` directory on your local computer, and run the command:
+
+```sh
+ansible-playbook -i inventory/hosts setup.yml --tags=stop
+```
+
+Alternatively, you can log in to the old server and run the command (you might have to `cd` to `/etc/systemd/system/` first):
+
+```sh
+systemctl disable --now matrix*
+```
```

Comment:
Only the `systemctl` commands makes sure services won't be starting again.

Running `--tags=stop` merely stops them once. If you reboot the server, they will start again.

Perhaps we should make a note of that.. and we'd probably better recommend running the `systemctl` command on the server as the better option. Maybe it should even be mentioned first.

---------

`cd`-ing into `/etc/systemd/system/` is also highly recommended, as the wildcard (`matrix*`) will likely not work properly otherwise.

Perhaps we should make the command be `cd /etc/systemd/system/ && systemctl disable --now matrix*` to make it easier to miss.